src/docs/testing/best-practices.md

```diff
@@ -0,0 +1,174 @@
+---
+title: Performance best practices
+short-title: Best practices
+description: How to ensure that your Flutter app is performant.
+---
+
+Generally Flutter applications are high performance by default,
+so you only need to avoid common pitfalls to get excellent performance
+instead of needing to micro-optimize with complicated profiling tools.
+These best recommendations will help you write the most performant
+Flutter app possible.
+
+## Best practices
+
+How do you design a Flutter app to most efficiently render your scenes?
+In particular, how do you ensure that the painting code generated by the
+framework is as efficient as possible? Here are a few things to consider
+when designing your app:
+
+### Call setState thoughtfully
```

Comment:
Perhaps surround `setState()` in back-ticks?

Reply:
Google Developer Style Guidelines says no codefont in titles/headings:
 https://developers.google.com/style/headings